### DISCUSSION TITLE
Test for overflow in puppyraffle

### DISCUSSION THREAD

**Discussion by guyzilberblum** [Cyfrin/security-and-auditing-full-course-s23#173] [General]
Hey i wrote a diffrante test from patrick test to proof an over flow let me know what you think 
"""

           function testOverFlows () public {
           address[] memory players = new address[](4);
           players[0] = playerOne;
           players[1] = playerTwo;
           players[2] = playerThree;
           players[3] = playerFour;
           puppyRaffle.enterRaffle{value: entranceFee * 4}(players);
    

           uint64 totalFees = 0;
           uint256 totalAmountCollected = players.length * entranceFee;
           uint256 prizePool = (totalAmountCollected * 80) / 100;
           uint256 fee = (totalAmountCollected * 20) / 100;
           uint64 totalfeesOverflow = 0;
           uint64 myVar = type(uint64).max;
           vm.warp(block.timestamp + duration + 1);
           vm.roll(block.number + 1);
           totalFees = totalFees + uint64(fee);
           totalfeesOverflow = totalfeesOverflow + myVar +1;

           puppyRaffle.selectWinner();
           console.log("max 64 uint", myVar);
           console.log("fee" ,fee);
           console.log("total fee", totalFees);
           console.log("this number should be zero" ,totalfeesOverflow);
        

    }
 
"""

let me know what you think

**Comment by EngrPips** [comment]
```javascript
function testOverFlows () public {
       address[] memory players = new address[](4);
       players[0] = playerOne;
       players[1] = playerTwo;
       players[2] = playerThree;
       players[3] = playerFour;
       puppyRaffle.enterRaffle{value: entranceFee * 4}(players);


       uint64 totalFees = 0;
       uint256 totalAmountCollected = players.length * entranceFee;
       uint256 prizePool = (totalAmountCollected * 80) / 100;
       uint256 fee = (totalAmountCollected * 20) / 100;
       uint64 totalfeesOverflow = 0;
       uint64 myVar = type(uint64).max;
       vm.warp(block.timestamp + duration + 1);
       vm.roll(block.number + 1);
       totalFees = totalFees + uint64(fee);
       totalfeesOverflow = totalfeesOverflow + myVar +1;

       puppyRaffle.selectWinner();
       console.log("max 64 uint", myVar);
       console.log("fee" ,fee);
       console.log("total fee", totalFees);
       console.log("this number should be zero" ,totalfeesOverflow);
}
```

**Reply by EngrPips** [reply]
I like the colors, which is why I re-formatted it; you can always format it as I did above by putting your block of code inside an upper three backticks (`) and a lower three backticks(`) and writing `javascript` or `js` immediately after the upper three backticks (`)

**Comment by guyzilberblum** [comment]
Thanks does the test is ok ?

**Reply by EngrPips** [reply]
When you run the test, did fees earned by the protocol overflow?

**Comment by guyzilberblum** [comment]
Basically I showed that if you add one to the maximum uint64 it will
overflow the fee is casted at the contract so it will overflow so basically
yea

**Reply by EngrPips** [reply]
Oh, I see, that seems correct, but doing it like that makes the quality of the finding subpar as you didn't make the fee of the protocol itself exceed the maximum value a uint64 can hold but instead just added the maximum value a uint64 can have to the current fee of the protocol. Regardless, you did a great job coming up with that.

**Comment by guyzilberblum** [comment]
Thanks a lot you helping a lot in my journey thanks for that

**Reply by EngrPips** [reply]
You are welcome, and I am glad you are learning and growing.
